### DISCUSSION TITLE
installation von ebusd auf einem Raspi mit home assistant image , das auf docker basiert

### DISCUSSION THREAD

**Discussion by KarLOs964cabrio** [john30/ebusd#631] [Q&A]
Hallo, ich versuche  ebusd zu installieren um meine Vaillant mit einer 630 Steuerung zu Üerwachen und erhalte dabei diesen Fehler.

Logger: homeassistant.setup
Source: components/ebusd/__init__.py:70
First occurred: 09:55:03 (1 occurrences)
Last logged: 09:55:03

Error during setup of component ebusd
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/ebusdpy/ebusdpy.py", line 14, in init
    sock.connect(address)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 235, in _async_setup_component
    result = await task
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/ebusd/__init__.py", line 70, in setup
    ebusdpy.init(server_address)
  File "/usr/local/lib/python3.10/site-packages/ebusdpy/ebusdpy.py", line 18, in init
    raise EBusError(socket.error)
ebusdpy.ebusdpy.EBusError: <class 'OSError'>

als ich versuchte die ensprechende datei zu finden war nicht mal der Pfad : python3.10 vorhanden.

Seltsamermerweise lässt sich aber der ebusd Dienst starten und empfängt einige daten , wie aussenthemperatur  und Datum .
alles andere kommen Fehler:

![image](https://user-images.githubusercontent.com/75625741/184870158-62aff601-f7f7-4798-941c-d81cc2a840e5.png)

gelegentlich kommen daten allerdings nur die vom broadcast.

Ich finde auch nicht die Stelle an der die csv/bai Dateien installiert sein sollen.

Kann mir bitte jemand sagen wie das richtig geht , denn die Anleitung von John30  kriege ich nicht mit einer Docker installation zum laufen. Ich weiss nicht mal welches BS in dem Dockercontainer läuft.

**Comment by Koky05** [comment]
I have same issue with "arbitration lost" but @john30 just answer to me it is in lattency, but nothing helps. I pick my lattency configuration up to 2000ms without any improvements. 

[#616](https://github.com/john30/ebusd/issues/616)

**Comment by john30** [comment]
please address issues with that HA integration to the corresponding author (not me).
a docker image usually does not contain a full OS but only parts of it. but that is irrelevant anyway as it is basically just an abstraction and isolation layer.
anyway, the CSVs are picked up from the web service unless you configure ebusd to do differently, see here: https://github.com/john30/ebusd/wiki/2.-Run#message-configuration-options
the quality of transfer depends a lot on the hwardware interface to eBUS, so e.g. when using one with a high latency, you need to adjust the ebusd start params accordingly and might still see issues like incomplete scanning as the communication does not work reliably

**Comment by LukasGrebe** [comment]
Hi, you're not experiencing a problem with the integration in Homeassistant but in unknown config data.

your clues here are:
1. you're receiving data from ebus, so it is up and running
2. the `received unknown MS cmd:` messages mean there is data on the ebus that the ebusdeamon can reformat into something readable and thus shows you the hex data of the ebus message.

regarding 2
this could be because the ebusd has not identified the ebus participants (scan config ERR:) - scan means ebusd is sending out data asking participants to identify themselves.

this identification is then used to load the correct "configuration files" from the internet to decode the received messages.

alternativ all bus participants have been identified but their messages are unknown/undocumented in the config files. then you need to get new messages defined. see https://github.com/john30/ebusd/wiki/HowTos

regarding (1) - there MAY be a problem with the hardware / latency thus you are getting `ERR: arbitration lost` wicht if I recall correctly is when ebusd wants to send out data for a scan but lost the bus arbitration.


lastly: the homeassistant integration is simply an interface between HA and a running ebusd. you still need to get ebusd up and running. you can do that in a docker container through home assistants docker container orchestrator called "supervisor". I built a homeassistant add-on to do exactly that. you can find out more at https://community.home-assistant.io/t/an-ebusd-add-on/344852
